src/media/peer.rs

```diff
@@ -58,6 +59,10 @@ impl PeerStateError {
 }
 
 /// Implementation of ['Peer'] state machine.
+#[state_machine_shared_fn_accessor(id -> Id)]
+#[state_machine_shared_fn_accessor(member_id -> MemberId)]
+#[state_machine_shared_fn_accessor(partner_peer_id -> Id)]
+#[state_machine_shared_fn_accessor(partner_member_id -> Id)]
```

Comment:
As for me, the result is to unintuitive. Reading this code I just can't understand what is happening at all.

After investing this quite a bit I've understood that the actual thing you're trying to do here is __delegation__ for enums. There are already several attempts for macros-based delegation in Rust out there: [delegate](https://docs.rs/delegate) and [enum_dispatch](https://docs.rs/enum_dispatch). However, as I see, they are not practical in this case. The problem with `delegate` that it's functional-style macro, so doesn't hold information about enum variants. And `enum_dispatch` requires some concrete "transparent" format for enums declaration and is strictly bound to traits, which is not the case here.

I think, that baking our own `#[delegate]` macro is a much better move here, as we don't bound the macro to a concrete/logic type, while gain readability and understandability benefits.

Let's compare with this design:
```rust
#[delegate(pub fn id(&self) -> Id)]
#[delegate(pub fn member_id(&self) -> MemberId)]
#[delegate(pub fn partner_peer_id(&self) -> Id)]
#[delegate(pub fn partner_member_id(&self) -> Id)]
```

Now it's way more intuitive what is going on, while the macro can be easily reused in other places.

The method signature should be specified the whole, because:
1. This gives far more control of what is going on for the caller. For example, the function can be static, or `self`, or `&mut self`, while current implementation is restricted to `&self` functions only. The other example may be the additional params of function.
2. This is much more readable and understandable for developer as we're actually declaring methods here.